### PR TITLE
update iam-contracts

### PR DESCRIPTION
**Pull request by JGiter**
Updated iam-contract exposes withdrawal delay and minimal staking period in StakingPool needed to calculate time left until withdrawal and request withdrawal became available

### CODE REVIEW

**Review by DawidGil-jlabs — APPROVED** [review]
looks good to me.

File: src/staking/index.ts [hunk 1]
```diff
@@ -103,20 +103,22 @@ export class StakingPoolService {
...
   }
 }
 
 /**
  * Abstraction over staking pool smart contract
  */
 export class StakingPool {
-  constructor(private pool: StakingPoolContract) { }
+  private pool: StakingPoolContract;
+  constructor(private patron: Required<Signer>, address: string) {
```
Comment (jrhender) [inline src/staking/index.ts:119]: Just curious, why the "Required<>" around the Signer? Does Signer have optional parameters that you want to ensure are set?

**Review by jrhender — APPROVED** [review]
Looks good

File: src/staking/index.ts (hunk 1 above)
Comment (JGiter) [reply src/staking/index.ts:119]: Correct. In order patron to send transaction he must be connected, but `provider` property of Signer is optional

**Pull request merged**